src/parser.py

```diff
@@ -1,158 +1,370 @@
+
+		#compare subcommand parser
+		compare_help_msg = "perform similarity comparsion with the use of fuzzy hashing"
+		self.parser_compare = self.subparsers.add_parser('compare', help= compare_help_msg, description = compare_help_msg)
+		self.parser_compare.add_argument('-fuzzy', metavar = 'FUZZY_HASH_FUNCTION_NAME', action = "store", help = "fuzzy hash function which will be used for the similarity comparsion")
+		self.parser_compare.add_argument('-t', '--targets', nargs='+', action = "store", metavar = "TARGETS_PARAMETER", required = True, help = "targets for which the hash value will be calculated")
+		self.parser_compare.add_argument('-dw', '--download-location', action = "store", metavar = "DOWNLOAD_LOCATION", default = getcwd(), help = "path to location where remote targets will be downloaded to. default: current working directory")
+		self.parser_compare.add_argument('-l', '--limit', action = "store", default = 1, type = int, metavar = "LIMIT", help = "only show results >= limit. default: 1")
+		self.parser_compare.add_argument('-r', '--recursive', action = "store_true", help = "allows recursive comparsion with the contents of directories")
+		self.parser_compare.add_argument('-o','--output', default="sys.stdout", action='store', metavar = "OUTPUT_PATH", help = "path to output file, default: stdout (Supported file formats: TXT, CSV, TSV, JSON, YAML, XML)")
+
+		comparsion_args = self.parser_compare.add_mutually_exclusive_group(required=True)
+		comparsion_args.add_argument('-a','--all',help='compare targets with all the files for which the respective fuzzy hash value is stored in the database')
+		comparsion_args.add_argument('-c','--comparsion-targets',nargs='+', action = "store", metavar = "FILE_ID", help='compare targets with a specific file, if ')
+
+	def parse(self, arg_list = None):
+		#parse the arguments and call the function that executes the correct subcommand
+		try:
+			args = self.parser.parse_args(arg_list)
+		except SystemExit as e:
+		#if the arguments are wrong, then don't exit 
+			print()
+		else:
+			try:
+				args.func(args)
+			except Exception as e:
+				print("Error: something went wrong during the execution of the command. In more detail:")
+				print(e)
+
+
+class TerminalParser(ParserTemplate):
+
+	'''This class inherits the parser of the ParserTemplate class and adds special arguments that are needed for the standalone (terminal)
+	commands only. It also impements the methods that call the App() methods with the correct parameters for each standalone command.'''
+
+	def __init__(self):
+
+		super(TerminalParser, self).__init__()
+		
+		##Terminal-only subcommands
+
+		self.parser_export.add_argument('-d', '--database', '--db', required = True, metavar = 'EXPORT_DATABASE_PATH', action = "store", help = "path to the new hashesdb database (.db file)")
```

Comment:
?!?

Reply:
When someone uses the standalone commands (through the terminal), then he has to specify a database with the use of the `-d` argument.
But when someone uses the REPL, he has to use the `use` command to select and "active" database.
For example, if someone wants to find out information about a database, he can use the standalone command
`hashesdb dbinfo -d mydatabase.db` or he can open the REPL, type 'use -d mydatabase.db' and then just write 'dbinfo'.